components/brave_rewards/browser/rewards_service_impl.cc

```diff
@@ -462,28 +462,44 @@ void RewardsServiceImpl::OnResult(
 
 void RewardsServiceImpl::MaybeShowBackupNotification(uint64_t boot_stamp) {
   PrefService* pref_service = profile_->GetPrefs();
-  bool user_has_funded = pref_service->GetBoolean(prefs::kRewardsUserHasFunded);
-  bool backup_succeeded = pref_service->GetBoolean(
-      prefs::kRewardsBackupSucceeded);
-  if (user_has_funded && !backup_succeeded) {
-    base::Time now = base::Time::Now();
-    base::Time boot_timestamp = base::Time::FromDoubleT(boot_stamp);
-    base::TimeDelta backup_notification_frequency =
-        pref_service->GetTimeDelta(prefs::kRewardsBackupNotificationFrequency);
-    base::TimeDelta backup_notification_interval =
-        pref_service->GetTimeDelta(prefs::kRewardsBackupNotificationInterval);
-    base::TimeDelta elapsed = now - boot_timestamp;
-    if (elapsed > backup_notification_interval) {
-      base::TimeDelta next_backup_notification_interval =
-          backup_notification_interval + backup_notification_frequency;
-      pref_service->SetTimeDelta(prefs::kRewardsBackupNotificationInterval,
-                                 next_backup_notification_interval);
-      RewardsNotificationService::RewardsNotificationArgs args;
-      notification_service_->AddNotification(
-          RewardsNotificationService::REWARDS_NOTIFICATION_BACKUP_WALLET, args,
-          "rewards_notification_backup_wallet");
+  base::TimeDelta interval = pref_service->GetTimeDelta(
+      prefs::kRewardsBackupNotificationInterval);
+
+  // Don't display notification if it has already been shown, if the
+  // balance is zero, or if a backup has already succeeded.
+  if (interval.is_zero() ||
+      !pref_service->GetBoolean(prefs::kRewardsUserHasFunded) ||
+      pref_service->GetBoolean(prefs::kRewardsBackupSucceeded)) {
+    return;
```

Comment:
Should this also zero out the notification interval as we do below with verified wallets?

Reply:
Good point - I've updated it so that the interval is cleared out if a backup has succeeded (but not if the user has zero balance).